src/frontend/components/node-item/node-item.ts

```diff
@@ -28,21 +26,23 @@ import {UserActionType}
           (click)="expandTree($event)">
         <div class="inline"
           [innerHTML]="getNodeDetails(node)">
-          </div>
+        </div>
       </div>
     
-      <div class="border-box pl4">
+      <div class="border-box pl4" *ngIf="node.isOpen || wasRendered">
```

Comment:
If a node was already rendered, don't remove it from the component tree if it is ever hidden. This will make viewing nodes in the component tree faster.